swiggy-pdp-cart/src/components/Main/Content/Content.js

```diff
@@ -0,0 +1,34 @@
+import React, { useEffect, useState } from "react";
+import Menu from "./Menu/Menu";
+import Cart from "./Cart/Cart";
+import Categories from "./Categories/Categories";
+import { fetchCategoriesList,fetchMenuList } from "../../../services/fakeFetchFunctions";
+import { categoryNameById } from "./helper/helper";
+import ContentCSS from './content.module.css'
+export default function Content(){
+  
+  const [category,setCategory]=useState('recommended');
+  const [menu,setMenu]=useState([]);
+  const [categoryList,setCategoryList]=useState([]);
+  
+  useEffect(() => {
+    const menu = fetchMenuList();
+    const categoryList = fetchCategoriesList();
+    setMenu(menu);
+    setCategoryList(categoryList);
+  },[]);  
+  const menuList = menu.filter((dish) => dish.categories.includes(category));
```

Comment:
Filter does not mutate the array and returns a new array reference each time.

Here, you are creating a new reference for `menuList` every time this component re renders, causing the reference to change which will unnecessarily re render `Menu` component as well.

Leverage `useMemo` here to preserve the reference across re renders.